Adds documentation for security demo configuration setup (#6058)
opensearch-project/documentation-website: _security/configuration/demo-configuration.md

---
layout: default
title: Setting up a demo configuration
parent: Configuration
nav_order: 4
---

# Setting up a demo configuration

> Welcome to the OpenSearch Security plugin's demo configuration setup guide. This tool provides a quick and easy way to replicate a production environment for testing purposes. The demo configuration includes the setup of security-related components such as internal users, roles, role mappings, audit configuration, basic authentication, tenants, and allow lists.

Reviewer: ```suggestion
Welcome to the OpenSearch Security plugin demo configuration setup guide. This tool provides a quick and easy way to replicate a production environment for testing purposes. The demo configuration includes the setup of security-related components, such as internal users, roles, role mappings, audit configuration, basic authentication, tenants, and allow lists.
```

The demo configuration tool performs the following tasks:

> 1. Configures security settings which are then loaded into the security index.

Reviewer: ```suggestion
1. Configures security settings, which are then loaded into the security index.
```

2. Generates demo certificates.

> 3. Updates the `opensearch.yml` file to add security-related settings.

Reviewer: ```suggestion
3. Adds security-related settings to the `opensearch.yml` file.
```

> ## Installing demo configuration

Reviewer: ```suggestion
## Installing the demo configuration
```

> The demo configuration is automatically called as part of the setup for each supported distribution of OpenSearch. Following are the instructions per distribution.

Reviewer: ```suggestion
The demo configuration is automatically called as part of the setup for each supported distribution of OpenSearch. The following are instructions for each distribution.
```

> **Note**: Starting from OpenSearch 2.12, a custom admin password is required to install demo configuration. If none is provided, the cluster would fail to start. Note, that this change affects fresh clusters only. Already setup clusters are not affected, as they already have `opensearch.yml` configured, and so the installation tool will not run. 

Reviewer: ```suggestion
**Note**: Starting with OpenSearch 2.12, a custom admin password is required in order to install the demo configuration. If none is provided, the cluster will fail to start. Note that this change only affects new clusters. Existing clusters are not affected because they already have `opensearch.yml` configured, so the installation tool will not run. 
```

### Docker

> Use the following steps to set up OpenSearch Security using Docker:

Reviewer: Should this be "to set up the Security plugin using Docker"?

> 1. Download [docker-compose.yml](https://opensearch.org/downloads.html)

Reviewer: ```suggestion
1. Download [docker-compose.yml](https://opensearch.org/downloads.html).
```

2. In the `docker-compose.yml` file, set `DISABLE_SECURITY_PLUGIN` to `false`.
3. Run the following command:

```bash
$ docker-compose up
```

### Setting up a custom admin password

> **Note**: If you are working with OpenSearch version 2.12 and later, it's crucial to set the initial admin password before installation. To customize the admin password, you can take the following steps:

Reviewer: ```suggestion
**Note**: For OpenSearch versions 2.12 and later, you must set the initial admin password before installation. To customize the admin password, you can take the following steps:
```

> **Note**: If you are working with OpenSearch version 2.12 and later, it's crucial to set the initial admin password before installation. To customize the admin password, you can take the following steps:

Reviewer: Should there be a line between the heading and the para?

1. Download the following sample [docker-compose.yml](https://github.com/opensearch-project/documentation-website/blob/{{site.opensearch_major_minor_version}}/assets/examples/docker-compose.yml) file.
2. Create a `.env` file.
3. Add the variable `OPENSEARCH_INITIAL_ADMIN_PASSWORD` and set the variable with a strong string password.
4. Run `docker-compose up`.

### TAR (Linux)

> For TAR distributions on Linux, download the the Linux set up files from the OpenSearch [downloads](https://opensearch.org/downloads.html) page. Then, use the following command to run the demo configuration: 

Reviewer: ```suggestion
For TAR distributions on Linux, download the Linux setup files from the OpenSearch [Download & Get Started](https://opensearch.org/downloads.html) page. Then use the following command to run the demo configuration: 
```

> For TAR distributions on Linux, download the the Linux set up files from the OpenSearch [downloads](https://opensearch.org/downloads.html) page. Then, use the following command to run the demo configuration: 

Reviewer: In most cases, we should use the actual name of the page.

```bash
$ ./opensearch-tar-install.sh
```

> For OpenSearch 2.12 or greater, set a new custom admin password before installation using the following command:

Reviewer: ```suggestion
For OpenSearch 2.12 or later, set a new custom admin password before installation by using the following command:
```

> For OpenSearch 2.12 or greater, set a new custom admin password before installation using the following command:

Reviewer: "running" instead of "using"?

```bash
$ export OPENSEARCH_INITIAL_ADMIN_PASSWORD=<custom-admin-password>
```

### Windows

> For ZIP distribution on Windows, after downloading and extracting:

Reviewer: ```suggestion
For ZIP distributions on Windows, after downloading and extracting the setup files, run the following command:
```

> For ZIP distribution on Windows, after downloading and extracting:

Reviewer: Confirm that my changes are technically accurate.

```powershell
> .\opensearch-windows-install.bat
```

> For OpenSearch 2.12 or greater, set a new custom admin password before installation using the following command:

Reviewer: ```suggestion
For OpenSearch 2.12 or later, set a new custom admin password before installation by using the following command:
```

> For OpenSearch 2.12 or greater, set a new custom admin password before installation using the following command:

Reviewer: "running" instead of "using"?

```powershell
> set OPENSEARCH_INITIAL_ADMIN_PASSWORD=<custom-admin-password>
```

### Helm

> For Helm charts, the demo configuration is automatically handled during the OpenSearch installation. For OpenSearch 2.12 and later, customize the admin password in `values.yaml` under `extraEnvs`:

Reviewer: ```suggestion
For Helm charts, the demo configuration is automatically handled during the OpenSearch installation. For OpenSearch 2.12 or later, customize the admin password in `values.yaml` under `extraEnvs`:
```

> For Helm charts, the demo configuration is automatically handled during the OpenSearch installation. For OpenSearch 2.12 and later, customize the admin password in `values.yaml` under `extraEnvs`:

Reviewer: Is there a more precise word than "handled"?

```yaml
extraEnvs:
  - name: OPENSEARCH_INITIAL_ADMIN_PASSWORD
    value: <custom-admin-password>
```

### RPM

> For RPM packages, install OpenSearch and set up the demo configuration using:

Reviewer: ```suggestion
For RPM packages, install OpenSearch and set up the demo configuration by running the following command:
```

> For RPM packages, install OpenSearch and set up the demo configuration using:

Reviewer: Confirm that my changes retain technical accuracy.

```bash
$ sudo yum install opensearch-{{site.opensearch_version}}-linux-x64.rpm
```

> For OpenSearch 2.12 or greater, set a new custom admin password before installation using the following command:

Reviewer: ```suggestion
For OpenSearch 2.12 or later, set a new custom admin password before installation by using the following command:
```

> For OpenSearch 2.12 or greater, set a new custom admin password before installation using the following command:

Reviewer: "running" instead of "using"?

```bash
$ sudo env OPENSEARCH_INITIAL_ADMIN_PASSWORD=<custom-admin-password> yum install opensearch-{{site.opensearch_version}}-linux-x64.rpm
```

### DEB

> For DEB packages, install OpenSearch and set up the demo configuration using:

Reviewer: ```suggestion
For DEB packages, install OpenSearch and set up the demo configuration by running the following command:
```

> For DEB packages, install OpenSearch and set up the demo configuration using:

Reviewer: Confirm that my changes retain technical accuracy.

```bash
$ sudo dpkg -i opensearch-{{site.opensearch_version}}-linux-arm64.deb
```

> For OpenSearch 2.12 or greater, set a new custom admin password before installation using the following command:

Reviewer: ```suggestion
For OpenSearch 2.12 or later, set a new custom admin password before installation by using the following command:
```

> For OpenSearch 2.12 or greater, set a new custom admin password before installation using the following command:

Reviewer: "running" instead of "using"?

```bash
$ sudo env OPENSEARCH_INITIAL_ADMIN_PASSWORD=<custom-admin-password> dpkg -i opensearch-{{site.opensearch_version}}-linux-arm64.deb
```

### Local distribution

> If you are building a local distribution, refer to the [DEVELOPER_GUIDE.md](https://github.com/opensearch-project/security/blob/main/DEVELOPER_GUIDE.md) for instructions on building a local binary for the Security plugin.

Reviewer: ```suggestion
If you are building a local distribution, refer to [DEVELOPER_GUIDE.md](https://github.com/opensearch-project/security/blob/main/DEVELOPER_GUIDE.md) for instructions on building a local binary for the Security plugin.
```

> For OpenSearch 2.12 or greater, make sure that a strong password is set before installation.

Reviewer: ```suggestion
For OpenSearch 2.12 or later, make sure that you set a strong password before installation.
```

---
layout: default

> title: Demo Configuration Setup Guide

Reviewer 2: Can we change this title to "Setting up a demo configuration" or something similar? That way it is consistent with the other items in the group under 'Configuration'.

Author: sounds good. I have kept your suggestion.

parent: Configuration
nav_order: 4
redirect_from:

>  - /security-plugin/configuration/demo-configuration/

Author: I would need doc-team's help in setting up proper navigation to this setup guide.

Reviewer 3: @DarshitChanpura: We likely don't need the redirect here since this is a new page, unless you are trying to redirect from a previously existing page.

Author: oh okay. I understood it as a path of the current page and that's why got confused on how it would be used. I can remove this.

---

# Demo configuration setup guide

> Welcome to the OpenSearch Security plugin's demo configuration setup guide. This tool provides a quick and easy way to replicate a production environment for testing purposes. The demo configuration includes the setup of security-related components such as internal users, roles, role mappings, audit configuration, basic authentication, tenants, and allowlists.

Reviewer 3: ```suggestion
The following guide provides provides a quick and easy way to replicate a production environment for testing purposes. The following demo configuration includes the setup of security-related components such as internal users, roles, role mappings, audit configuration, basic authentication, tenants, and allow lists.
```

> ## Overview

Reviewer 3: ```suggestion

```

The demo configuration tool performs the following tasks:

1. Configures security settings which are then loaded into the security index.
2. Generates demo certificates.
3. Updates the `opensearch.yml` file to add security-related settings.

## Installing demo configuration

The demo configuration is automatically called as part of the setup for each supported distribution of OpenSearch. Following are the instructions per distribution.

> **Note**: Starting from OpenSearch 2.12, a custom admin password is required to install demo configuration. If none is provided, the cluster would fail to start. Please note, that this change affects fresh clusters only. Already setup clusters are not affected, as they already have `opensearch.yml` configured, and so the installation tool will not run. 

Reviewer 3: ```suggestion
**Note**: As of OpenSearch 2.12, a custom admin password is required to install demo configuration. If none is provided, the cluster will not start. This change only affects new OpenSearch clusters with OpenSearch 2.12 or greater.
```

### Docker

> Follow these steps to set up OpenSearch Security using Docker:

Reviewer 3: ```suggestion
Use the following steps to set up OpenSearch Security using Docker:
```

> 1. [Download docker-compose.yml](https://opensearch.org/downloads.html)

Reviewer 3: ```suggestion
1. [Download docker-compose.yml](https://opensearch.org/downloads.html).
```

> 2. Ensure that the `DISABLE_SECURITY_PLUGIN` environment variable is not set to true.

Reviewer 3: ```suggestion
2. In the `docker-compose.yml` file, set `DISABLE_SECURITY_PLUGIN` to `false`.
```

> 3. Run the following command in your terminal:

Reviewer 3: ```suggestion
3. Run the following command:

```

```bash
$ docker-compose up
```

> 

Reviewer 3: ```suggestion

### Setting up a custom admin password
```

**Note**: If you are working with OpenSearch version 2.12 and later, it's crucial to set the initial admin password before installation. To customize the admin password, you can take the following steps:

> - Download the sample [docker-compose.yml](https://github.com/opensearch-project/documentation-website/blob/{{site.opensearch_major_minor_version}}/assets/examples/docker-compose.yml).

Reviewer 3: ```suggestion
1. Download the following sample [docker-compose.yml](https://github.com/opensearch-project/documentation-website/blob/{{site.opensearch_major_minor_version}}/assets/examples/docker-compose.yml) file.
```

> - Create a `.env` file.

Reviewer 3: ```suggestion
2. Create a `.env` file.
```

Reviewer 3: What directory does the `.env` file live?

> - Add the `OPENSEARCH_INITIAL_ADMIN_PASSWORD` variable with a strong password.

Reviewer 3: ```suggestion
3. Add the variable `OPENSEARCH_INITIAL_ADMIN_PASSWORD` and set the variable with a strong string password.
```

> - Run `docker-compose up`.

Reviewer 3: ```suggestion
4. Run `docker-compose up`.
```

### TAR (Linux)

> For TAR distribution on Linux, after downloading and extracting:

Reviewer 3: ```suggestion
For TAR distributions on Linux, download the the Linux set up files from the OpenSearch [downloads](https://opensearch.org/downloads.html) page. Then, use the following command to run the demo configuration: 
```

```bash
$ ./opensearch-tar-install.sh
```

> **Note**: For OpenSearch 2.12 and later, set the initial admin password before installation:

Reviewer 3: ```suggestion
For OpenSearch 2.12 or greater, set a new custom admin password before installation using the following command:
```

```bash

> $ export OPENSEARCH_INITIAL_ADMIN_PASSWORD=< Admin password >

Reviewer 3: ```suggestion
$ export OPENSEARCH_INITIAL_ADMIN_PASSWORD=<custom-admin-password>
```

```

### Windows

For ZIP distribution on Windows, after downloading and extracting:

```powershell
> .\opensearch-windows-install.bat
```

> **Note**: For OpenSearch 2.12 and later, set the initial admin password before installation:

Reviewer 3: ```suggestion
For OpenSearch 2.12 or greater, set a new custom admin password before installation using the following command:
```

```powershell

> > set OPENSEARCH_INITIAL_ADMIN_PASSWORD=< Admin password >

Reviewer 3: ```suggestion
> set OPENSEARCH_INITIAL_ADMIN_PASSWORD=<custom-admin-password>
```

```

### Helm

For Helm charts, the demo configuration is automatically handled during the OpenSearch installation. For OpenSearch 2.12 and later, customize the admin password in `values.yaml` under `extraEnvs`:

```yaml
extraEnvs:
  - name: OPENSEARCH_INITIAL_ADMIN_PASSWORD
    value: < Admin password >
```

### RPM

For RPM packages, install OpenSearch and set up the demo configuration using:

```bash
$ sudo yum install opensearch-{{site.opensearch_version}}-linux-x64.rpm
```

> **Note**: For OpenSearch 2.12 and later, set the initial admin password before installation:

Reviewer 3: ```suggestion
For OpenSearch 2.12 or greater, set a new custom admin password before installation using the following command:
```

```bash

> $ sudo env OPENSEARCH_INITIAL_ADMIN_PASSWORD=< Admin password > yum install opensearch-{{site.opensearch_version}}-linux-x64.rpm

Reviewer 3: ```suggestion
$ sudo env OPENSEARCH_INITIAL_ADMIN_PASSWORD=<custom-admin-password> yum install opensearch-{{site.opensearch_version}}-linux-x64.rpm
```

```

### DEB

For DEB packages, install OpenSearch and set up the demo configuration using:

```bash
$ sudo dpkg -i opensearch-{{site.opensearch_version}}-linux-arm64.deb
```

> **Note**: For OpenSearch 2.12 and later, set the initial admin password before installation:

Reviewer 3: ```suggestion
For OpenSearch 2.12 or greater, set a new custom admin password before installation using the following command:
```

```bash

> $ sudo env OPENSEARCH_INITIAL_ADMIN_PASSWORD=< Admin password > dpkg -i opensearch-{{site.opensearch_version}}-linux-arm64.deb

Reviewer 3: ```suggestion
$ sudo env OPENSEARCH_INITIAL_ADMIN_PASSWORD=<custom-admin-password> dpkg -i opensearch-{{site.opensearch_version}}-linux-arm64.deb
```

```

### Local distribution

If you are building a local distribution, refer to the [DEVELOPER_GUIDE.md](https://github.com/opensearch-project/security/blob/main/DEVELOPER_GUIDE.md) for instructions on building a local binary for the Security plugin.

> **Note**: For OpenSearch 2.12 and later, ensure that a strong password is set before installation.

Reviewer 3: ```suggestion
For OpenSearch 2.12 or greater, make sure that a strong password is set before installation.
```
